bin/collator/src/rpc.rs

```diff
@@ -407,3 +373,133 @@ where
 
     Ok(io)
 }
+
+struct AuraConsensusDataProviderFallback<B, C> {
```

Comment:
Suggestion - perhaps having this together with the _migration_ type already used by SHiden is a good idea. Right now they are scattered between different files.

Request - please document this well, and explain the underlying issue so we don't need to research about this again, if issue re-appears due to some future change. 🙂